blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx

```diff
@@ -80,12 +80,34 @@ export class RunDetailsPipeline extends Component {
     }
 
     componentWillReceiveProps(nextProps) {
-        if (this.props.result.isQueued()) {
+        if (nextProps.result.isQueued()) {
             return;
         }
         const followAlong = this.state.followAlong;
         this.mergedConfig = this.generateConfig({ ...nextProps, followAlong });
-
+        
+        if (this.mergedConfig.supportsNode && (!nextProps.nodes || !nextProps.nodes[this.mergedConfig.nodeKey])) {
+            const tryNodes = () => {
+                // in order to handle queued items during re-run, we need to retry fetching nodes periodically
+                if (!this.props.nodes || !this.props.nodes[this.mergedConfig.nodeKey]) {
```

Comment:
Hmm could we not relay on SSE events to fetch when needed? 

Using from  https://github.com/jenkinsci/blueocean-plugin/pull/638/files#diff-67a79c3214f72b25c36381b96fdbebc0 e.g.  `pipeline_start` event and then in the case add

```
            case 'pipeline_start':
                {
                    delete this.mergedConfig.node;
                    fetchNodes({ ...this.mergedConfig, refetch });
                }
```

That would handle start events.